### PR TITLE
Replaces Defiler's Neurotoxin sting with Reagent Slash, adds associated toxins, adds 2 new gasses, adds hud for xenos based on new toxins V2

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request

Replaces Defiler's Neurotoxin Sting with Reagent Slash that can cycle between neurotoxin and 2 new toxins.

 - Reagent Slash - 100 plasma, 6s CD - slashes 4 times with 1.2s delay between slashes. Each slash applies 3u of selected reagent and deals 6-9 brute damage (based on age)
 - Neurogas ability is able to emit 2 additional types of smoke that are transparent and apply the 2 new reagents. Emitted gas is based on selected reagent

New reagents:
these reagents have the same metabolism as neurotoxin, with 12u lasting 60s without being purged

 - Transvitox - after ~4s, if the target has less than 45 tox damage, when damage is taken converts brute/burn to toxin damage equal to 40% of the damage amount
 - Hemodile - slows movement speed by 25% or 50% with neurotoxin and deals stamina damage based on 20% of damage taken

For anyone wondering why this is reposted, I messed up original pull request and made this one with the latest intended changes

## Why It's Good For The Game

This adds a new combat ability to Defiler and 2 more smoke variants neurogas, giving it more options when fighting and making it a more interesting caste to evolve into.

## Changelog
:cl: Slotbot
add: Added reagent slash and select reagent abilities
add: Added hemodile and transvitox toxins
add: Added hud indicators over marines showing said toxins for xenos
add: Added smoke versions of new reagents, emitted by having the reagent selected and activating neurogas
del: Removed neutoroxin sting from Defiler
imageadd: added reagent hud icons
/:cl:

### CODE REVIEW

**Review by TiviPlus — APPROVED** [review]

**Comment by Isy232** [comment]
So transvitox actually decreases the amount of damage marines take

**Comment by Walarks** [comment]
> So transvitox actually decreases the amount of damage marines take

It gives toxin damage which is harder to heal and causes a stun at 45. It is situationally good in this regard.

File: code/modules/reagents/reagents/toxin.dm [hunk 1]
```diff
@@ -518,3 +518,61 @@
...
+	if(prob(25))
+		to_chat(L, "<span class='warning'>You feel your legs tense up.</span>")
+	if(!L.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_neurotoxin))
+		L.add_movespeed_modifier(MOVESPEED_ID_XENO_HEMODILE, TRUE, 0, NONE, TRUE, 1)
+	else
+		L.add_movespeed_modifier(MOVESPEED_ID_XENO_HEMODILE, TRUE, 0, NONE, TRUE, 3)
+	return ..()
+
+/datum/reagent/toxin/xeno_hemodile/proc/hemodile_human_damage_taken(mob/living/L, damage)
+	L.adjustStaminaLoss(damage*0.2)
```
Comment (TiviPlus) [inline code/modules/reagents/reagents/toxin.dm:545]: signal handler

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 2]
```diff
@@ -155,9 +163,86 @@
...
+		return FALSE
+	if(!A.can_sting())
+		to_chat(owner, "<span class='warning'>Our slash won't affect this target!</span>")
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/reagent_slash/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	succeed_activate()
+	slash_action(A, X.selected_reagent, channel_time = DEFILER_REAGENT_SLASH_DELAY, count = DEFILER_REAGENT_SLASH_COUNT, transfer_amount = DEFILER_REAGENT_SLASH_U_AMOUNT)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:228]: No need for the named arguments here

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 3]
```diff
@@ -155,9 +163,86 @@
...
+	button.overlays += image('icons/mob/actions.dmi', button, initial(A.name))
+	return ..()
+
+/datum/action/xeno_action/select_reagent/action_activate()
+	var/mob/living/carbon/xenomorph/X = owner
+	var/list/available_reagents = X.xeno_caste.available_reagents_define
+	var/i = available_reagents.Find(X.selected_reagent)
+	if(length(available_reagents) == i)
+		X.selected_reagent = available_reagents[1]
+	else
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:195]: This code is confusing and doesnt work properly

Comment (Walarks) [reply code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:195]: Can you elaborate more on how it is not working? It is switching reagents as intended on my end.

**Comment by Dolth** [comment]
This is a major tweak to one of the most under appreciated caste.
Also if you could put verbs to allow macros or just regular keybinding abilities for the reagent slash/reagent selection it'd be great.
Without it it's pretty hard to do the neuro/hemo combo.

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm (hunk 3 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:195]: var/i = available_reagents.Find(X.selected_reagent)
	if(length(available_reagents) == i)
		X.selected_reagent = available_reagents[1]

find the selected reagent and set it to i
if the length of available reagents == reference to the selected reagent
select the first reagent in the list

This should be length == 1, i should be removed or renamed (?)

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**